compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -249,7 +249,7 @@ impl<'mir, 'tcx> Checker<'mir, 'tcx> {
         let secondary_errors = mem::take(&mut self.secondary_errors);
         if self.error_emitted.is_none() {
             for error in secondary_errors {
-                error.emit();
+                self.error_emitted = Some(error.emit());
```

Comment:
@RalfJung found the issue that caused us to const eval even though there were const qualif errors 🙈